metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -1020,10 +1044,12 @@ def _find_dataflow_recipe(
             metric_time_dimension_reference=self._metric_time_dimension_reference,
```

Comment:
Update this function to do the conversion from custom to base grain, keep that change contained inside it
and remove related logic from elsewhere